_meeting-notes/2019-12-03.md

```diff
@@ -0,0 +1,168 @@
+    - AI: Steven will finish the DHT Refactor PR
+- ASKs
+    - @daviddias - ASK: Please use Kanban board to communicate effectively the state of your issues/PRs. I keep seeing PRs in the "New Issues" column and wondering if you want review or not. If you haven't, read: https://github.com/ipfs/testground/blob/master/docs/HOW_WE_WORK.md
+        -~~ AI: make this discoverable! (David)~~ (moved to Assignments, below)
+    - @raulk - ASK: to @stebalien: could you write a design blueprint for traffic shaping? Let's make the current behaviour the starting point. Further enhancements / changes would be patches on top of that doc.
+        - ~~YES! AI: Steven said he can do it. <- push it to the docs folder~~ (moved to Assignments, below)
+
+
+## :dart: Next
+
+~~### OKRs for the Quarter~~ #1 Goal: Unblock go-ipfs 0.5.0
+~~https://raw.githubusercontent.com/ipfs/testground/master/docs/img/Q4OKRs.png~~
+~~### Top 3 things to get done this week~~
+
+
+
+We are putting aside the OKRs for the time being and instead **focus on the #1 Goal: Unblock go-ipfs 0.5.0**. This goal has a very clear list of things that need to ship:
+
+- [ ] Instrument go-libp2p so that the DHT operations are observable and traced in a Distributed Context (owner: @raulk)
+- [ ] Add support to Traffic Shaping & Network Configuration to TestGround (owner: @Stebalien)
+- [ ] Expand the TestGround Infrastructure so that it can run multiple thousands of nodes (owner: @nonsense)
+- [ ] Creation of a TestGround Dashboard that can aid the developers of go-ipfs & go-libp2p to grok the results outputted by every node in a TestGround run (owner: @daviddias)
+- [ ] Keep on iterating the DHT Test Plan (@owner: @daviddias)
+- [ ] Design blueprint for multi-version tests (aka test ensembles) (owner: @raulk)
+
+Full list can be at https://github.com/ipfs/testground/issues/196
+
+
+### Assignments
+
+- ==@raulk==
+    - Review daemon PR.
+- ==@daviddias==
+    - I'll have to dedicate most of my week to:
+        - Life Admin
+        - ResNetLab Deliverables (so that I can be fully heads down during the Hack Week)
+    - For TestGround specifically
+        - Continue preparing the Hack Week
+        - Collect architecture decisions that need to be captured and written down
+        - Create an Epic that tracks the Dependency Graph Tracking and link it to the go-ipfs 0.5.0 epic
+        - Link to HOW_WE_WORK from CONTRIBUTING.md
+- ==@stebalien==
+    - go-ipfs Hack Week
+    - For TestGround
+        - Last few changes on sidecar (the jitter PR + blocking inbound connection)
+        - Follow up on Jim's suggestion for Network Isolation (NAT) https://github.com/ipfs/testground/issues/96#issuecomment-559888146
+        - Write a Design Blueprint for Traffic Shapping (take inspiration from https://github.com/ipfs/testground/pull/203)
+- ==@jimpick==
+    - Test Plan 4
+    - Test SideCar and demonstrate how to use it (docs or recorded demo)
+- ==@nonsense==
```

Comment:
@nonsense similar to @jimpick's, can you unpack a bit more on what is your plan for the Week? I believe you have an issue coming up describing the "TestGround as a Service" vision, opening up for discussion (shared resource vs. a cluster for everyone)

Reply:
@daviddias updated the doc with a bit more granularity on the work to be done on `infra`. Not sure what you mean with the `TestGround as a Service` vision issue... I looked for this in the Github Issues, but I am not sure which one you're referring to...

Reply:
Ah! fair point. It was indeed a note from our 1:1 back from Monday, perhaps it wasn't clear enough. The ask & recommendation made at the time was for you to feel empowered to own painting the picture and paving the path for what "TestGround as a Service" will be. In "my world view" that task would start as an issue and hence, me mentioning "an issue"